src/Workspaces/Core/Portable/Classification/ClassifierHelper.cs

```diff
@@ -90,19 +101,25 @@ private static void RemoveAdditiveSpans(ArrayBuilder<ClassifiedSpan> spans)
             AdjustSpans(syntaxSpans, widenedSpan);
             AdjustSpans(semanticSpans, widenedSpan);
 
-            // The classification service will only produce classifications for
-            // things it knows about.  i.e. there will be gaps in what it produces.
-            // Fill in those gaps so we have *all* parts of the span 
-            // classified properly.
-            using var _1 = ArrayBuilder<ClassifiedSpan>.GetInstance(out var filledInSyntaxSpans);
-            using var _2 = ArrayBuilder<ClassifiedSpan>.GetInstance(out var filledInSemanticSpans);
-
-            FillInClassifiedSpanGaps(widenedSpan.Start, syntaxSpans, filledInSyntaxSpans);
-            FillInClassifiedSpanGaps(widenedSpan.Start, semanticSpans, filledInSemanticSpans);
+            if (fillInClassifiedSpanGaps)
+            {
+                // The classification service will only produce classifications for
+                // things it knows about.  i.e. there will be gaps in what it produces.
+                // By default, fill in those gaps so we have *all* parts of the span 
+                // classified properly.
+                using var _1 = ArrayBuilder<ClassifiedSpan>.GetInstance(out var filledInSyntaxSpans);
+                using var _2 = ArrayBuilder<ClassifiedSpan>.GetInstance(out var filledInSemanticSpans);
+
+                FillInClassifiedSpanGaps(widenedSpan.Start, syntaxSpans, filledInSyntaxSpans);
+                FillInClassifiedSpanGaps(widenedSpan.Start, semanticSpans, filledInSemanticSpans);
+
+                syntaxSpans = filledInSyntaxSpans;
+                semanticSpans = filledInSemanticSpans;
```

Comment:
this is *definitely* not going to work.  :)  remember that when the using goes out of scope, we'll put the array back in teh pool.  meaning someone else might grab on it and party on it at the same time as us.  i have a recommendtation on how to structure this code.  ping me :)